cdap-docs/admin-manual/source/security/authorization.rst

```diff
@@ -354,103 +347,238 @@ Kerberos Principal
+| Creating a dataset                           |                                                                                                                         | *ADMIN* on the dataset                                                                                                  |
++----------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+
+| Creating a stream                            |                                                                                                                         | *ADMIN* on the stream                                                                                                   |
++----------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+
+| Creating a custom dataset during deployment  |                                                                                                                         | *ADMIN* on the new dataset module and type (use the full class name of the custom dataset as the module id and type id) |
++----------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+
+| Using an existing custom dataset             |                                                                                                                         | *ADMIN* on the existing dataset module and type                                                                         |
++----------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------+
+
+.. _security-authorization-executing-programs:
+
+Execute Programs/Hydrator Pipelines
+-----------------------------------
+To execute a program or a pipeline, the requesting user will need *EXECUTE* privilege on it. If there is no impersonation,
+the program will run (executing user) as the CDAP master user. If impersonation is involved, the program will run
```

Comment:
It is a little confusing to have `(executing user)` here, can we remove it?

Reply:
We need to define the executing user here so that we can use it later.

Reply:
👍